File: src/main/java/com/wanted/preonboarding/domain/product/dto/response/ProductResponse.java
package com.wanted.preonboarding.domain.product.dto.response;

import com.wanted.preonboarding.domain.product.entity.Product;
import com.wanted.preonboarding.domain.product.entity.ProductState;
import com.wanted.preonboarding.domain.purchase.entity.PurchaseState;
import com.wanted.preonboarding.domain.user.dto.response.UserResponse;
import lombok.*;

@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ProductResponse {
    private Long id; // 아이디
    private UserResponse seller; // 판매자
    private String name; // 제품명
    private Integer price; // 가격
    private Integer quantity; // 재고
    private ProductState state; // 상태

    public static ProductResponse of(Product product) {
        return ProductResponse.builder()
                .id(product.getId())
                .seller(UserResponse.of(product.getUser()))
                .name(product.getName())
                .price(product.getPrice())
                .quantity(product.getQuantity())
                .state(product.getState())
                .build();
    }

}

File: src/main/kotlin/wanted/market/product/dto/ProductResponse.kt
package wanted.market.product.dto

import wanted.market.product.entity.Product

data class ProductResponse(val name: String,
                      val price: Int,
                      val productStatus: String) {

    companion object {
        fun from(product: Product): ProductResponse {
            return ProductResponse(
                product.priceName,
                product.price,
                product.productStatus.value)
        }
    }
}